src/storage/mvcc/reader/forward_seeker.rs

```diff
@@ -152,69 +153,61 @@ impl<S: Snapshot> ForwardSeeker<S> {
         &self.statistics
     }
 
-    /// Get the next key-value pair, where the key is greater or equal to given `key`.
-    pub fn read_next(&mut self, mut key: Key, ts: u64) -> Result<Option<(Key, Value)>> {
-        if key < self.last_read_key {
-            panic!(
-                "ForwardSeeker: try to read from key {} which is less than previous read key {}",
-                key, self.last_read_key
-            );
+    /// Get the next key-value pair.
+    pub fn read_next(&mut self, ts: u64) -> Result<Option<(Key, Value)>> {
```

Comment:
I believe that ts can be also removed from the parameter, because for a scanning process, we are dealing with the same ts and continuously iterate next.

Reply:
Oh I thought it too but forgot to remove it. sorry.